.cirrus.yml

```diff
@@ -1,25 +1,37 @@
-root_task_template: &ROOT_TASK_TEMPLATE
-  # Don't run on release tags since it creates O(n^2) tasks where n is the
-  # number of plugins
-  only_if: $CIRRUS_TAG == ''
-  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
-  env:
-    INTEGRATION_TEST_PATH: "./packages/integration_test"
-    CHANNEL: "master" # Default to master when not explicitly set by a task.
-    PLUGIN_TOOLS: "dart run ./script/tool/lib/src/main.dart"
-  setup_script:
-    - flutter channel $CHANNEL
-    - flutter upgrade
+setup_tool_template: &SETUP_TOOL_TEMPLATE
+  setup_tool_script:
     - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
     - cd script/tool
     - pub get
 
+flutter_upgrade_template: &FLUTTER_UPGRADE_TEMPLATE
+  upgrade_flutter_script:
+    - flutter channel $CHANNEL
+    - flutter upgrade
+  << : *SETUP_TOOL_TEMPLATE
+
+macos_template: &MACOS_TEMPLATE
+  # Only one macOS task can run in parallel without credits, so use them for
+  # PRs on macOS.
+  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
+  osx_instance:
+    image: big-sur-xcode-12.3
+  cocoapod_install_script: sudo gem install cocoapods
+
+# Don't run on release tags since it creates O(n^2) tasks where n is the
+# number of plugins
+only_if: $CIRRUS_TAG == ''
+use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
+env:
+  INTEGRATION_TEST_PATH: "./packages/integration_test"
+  CHANNEL: "master" # Default to master when not explicitly set by a task.
+  PLUGIN_TOOLS: "dart run ./script/tool/lib/src/main.dart"
```

Comment:
@stuartmorgan I thought of this right after you merged #3697.  These common properties can be at the top level, they don't need to be in an included duplicate node or under a task.

Reply:
🤦 

I saw that change in your earlier PR, but completely forgot.

Reply:
Nit: I think it would read more naturally if this were first, before the templates that in some cases override things here.

Reply:
Done.